### PR TITLE
Update and clarify preview instructions.

### PR DESCRIPTION
**Pull request by mbrukman**
Provide more details and clarity about how to submit changes for this
repo and how to make a preview of the website to enable sharing with
reviewers and other observers.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,30 +42,92 @@ Before pushing the site to production, and a great way to demonstrate proposed
...
+   $ cd janusgraph.org
+   $ git remote add upstream git@github.com:JanusGraph/janusgraph.org.git
+   ```
+
+1. Create a new branch for your changes, based on the latest changes in the
+   `master` branch of the upstream repo:
+
+   ```
+   $ git checkout master
+   $ git pull upstream master
```
Comment (hsaputra) [inline README.md]: Assuming my clone is old one, we should also do `git merge upstream/master` to make sure local `master` has latest from the upstream repo.

**Comment by hsaputra** [comment]
As far as I know doing `git pull upstream master` will only update your
local upstream/master copy.

Need to merge to your fork of origin/master to sync with upstream:

https://help.github.com/articles/syncing-a-fork/

Thanks,

- Henry

**Comment by mbrukman** [comment]
@hsaputra — what I typically do is:

Bring local `master` to match upstream:

```
$ git checkout master
$ git pull upstream master
```

If I want my GitHub fork's `master` to be up-to-date (which is really optional, but why not):

```
$ git push    # automatically pushes local `master` to `origin/master`
```

If I want to update my feature branch:

```
$ git checkout my-feature-branch
$ git rebase master
$ git push -f     # update my PR or in-progress branch to have a clean diff
```

If I merge `master` into my feature branch, it becomes more complex and has a lot of extra spurious commits, whereas I want it to only have a single commit (or 2 if that's what I'm trying to do specifically).

The rebasing of my feature branch on the updated `master` is described lower in the README.

Does that address your concern, or were you looking for something else?

**Comment by hsaputra** [comment]
Ohhh yes, you are right. That will work too.

I miss where it is checkout the origin master branch.

Will update my review, thx

- Henry

File: README.md [hunk 2]
```diff
@@ -42,30 +42,93 @@ Before pushing the site to production, and a great way to demonstrate proposed
...
-1. checkout the `gh-pages` branch or create it if it doesn't exist
-1. make changes on that branch and push `gh-pages` to your fork on GitHub
-1. visit the "Settings" page for your fork on GitHub and make the `gh-pages`
+
+1. Clone your copy of the repo locally
+
+   ```
+   $ git clone git@github.com:[USERNAME]/janusgraph.org.git
+   $ cd janusgraph.org
+   $ git remote add upstream git@github.com:JanusGraph/janusgraph.org.git
```
Comment (hsaputra) [inline README.md]: Dont you need to do `git fetch upstream` right after this?

**Comment by hsaputra** [comment]
@mbrukman Added one last question/ comment on the PR.

File: README.md (hunk 2 above)
Comment (hsaputra) [reply README.md]: Never mind, will be pulled later.

**Comment by mbrukman** [comment]
@hsaputra — I also remembered and added the use of `--ff-only` flag to ensure that `master` is a clean copy of upstream and added some documentation about this. Please take a look and let me know what you think. Thanks for the review!

**Comment by mbrukman** [comment]
@hsaputra — is the PR still OK to merge? See my most recent comment above for changes made since your last review. Let me know if you have any concerns before I merge. Thanks!

**Comment by hsaputra** [comment]
@mbrukman Yeah, LGTM

Sorry I didn't make it clear.

**Comment by mbrukman** [comment]
Thanks, @hsaputra and @amcp, for the reviews!

**Pull request merged**